Lib/multiprocessing/connection.py

```diff
@@ -275,19 +274,27 @@ class PipeConnection(_ConnectionBase):
         _got_empty_message = False
         _send_ov = None
 
-        def _close(self, _CloseHandle=_winapi.CloseHandle):
+        def _close(self, handle, *, _CloseHandle=_winapi.CloseHandle):
             ov = self._send_ov
             if ov is not None:
                 # Interrupt WaitForMultipleObjects() in _send_bytes()
                 ov.cancel()
-            _CloseHandle(self._handle)
+            _CloseHandle(handle)
 
         def _send_bytes(self, buf):
             if self._send_ov is not None:
                 # A connection should only be used by a single thread
                 raise ValueError("concurrent send_bytes() calls "
                                  "are not supported")
-            ov, err = _winapi.WriteFile(self._handle, buf, overlapped=True)
+            try:
+                ov, err = _winapi.WriteFile(self._handle, buf, overlapped=True)
+            except OSError as err:
+                if err.errno == errno.EBADF:
+                    raise OSError(errno.EPIPE, "handle is closed")
+                raise
+            except TypeError:
+                self._check_closed()
+                raise
```

Comment:
Would it be possible to split this change in two parts? Keep this change for _check_closed() change which now raises BrokenPipeError, and you changes around close(): these ones look straightforward are correct. But write a separated PR to add these try/except?

I dislike these try/except. If there is a risk to land into EBADF case with the current code, maybe some kind of locking is needed to *not* call the Windows API with a handle, while another thread call close() which invalidates the handle.